### PR TITLE
Enable flake8-async rule

### PR DESCRIPTION
**Pull request by CoolCat467**
This PR enables the flake8-async rule and fixes all the existing issues that were raised.

### CODE REVIEW

File: trio/_tests/type_tests/check_wraps.py [hunk 1]
```diff
@@ -1,8 +1,7 @@
 # https://github.com/python-trio/trio/issues/2775#issuecomment-1702267589
 # (except platform independent...)
-import typing_extensions
-
 import trio
+import typing_extensions
 
 
 async def fn(s: trio.SocketStream) -> None:
```
Comment (A5rocks) [inline trio/_tests/type_tests/check_wraps.py:7]: Above got fixed on master btw

File: notes-to-self/loopy.py [hunk 2]
```diff
@@ -6,7 +6,7 @@
 async def loopy():
     try:
         while True:
-            time.sleep(0.01)
+            time.sleep(0.01)  # noqa: ASYNC101  # OS sleep so no CPU fire
```
Comment (A5rocks) [inline notes-to-self/loopy.py:9]: I feel like the comment can be reworded slightly :^)

**Review by A5rocks — APPROVED** [review]
Looks good, nitpicks

File: notes-to-self/loopy.py (hunk 2 above)
Comment (CoolCat467) [reply notes-to-self/loopy.py]: Re-worded comment in 1b9643d2

**Review by A5rocks — APPROVED** [review]

**Pull request merged**